assets/js/blocks/cart-checkout/cart/edit.js

```diff
@@ -38,17 +38,8 @@ const BlockSettings = ( { attributes, setAttributes } ) => {
 		isShippingCostHidden,
 		checkoutPageId,
 	} = attributes;
-	const pages =
-		useSelect( ( select ) => {
-			return select( 'core' ).getEntityRecords( 'postType', 'page', {
-				status: 'publish',
-				orderby: 'title',
-				order: 'asc',
-				per_page: 100,
-			} );
-		}, [] ) || null;
 	const { currentPostId } = useEditorContext();
-
+	const { current: savedCheckoutPageId } = useRef( checkoutPageId );
```

Comment:
We save the supplied (saved) value in a ref and use it, this will ensure it will get updated when changed, another method could have been
```
const [ savedCheckoutPageId ] = useState( checkoutPageId );
```
But this pattern is more readable and known.